rand_core/src/impls.rs

```diff
@@ -109,4 +110,39 @@ pub fn next_u128_via_fill<R: Rng+?Sized>(rng: &mut R) -> u128 {
     impl_uint_from_fill!(rng, u128, 16)
 }
 
+const LIMIT_ERR_MAX: usize = 20;    // arbitrary
+static LIMIT_ERR_COUNT: AtomicUsize = ATOMIC_USIZE_INIT;
+
+/// Implement `fill_bytes` via `try_fill` with implicit error handling.
+pub fn fill_via_try_fill<R: Rng+?Sized>(rng: &mut R, dest: &mut [u8]) {
+    loop {
+        if let Err(e) = rng.try_fill(dest) {
+            if e.kind.should_retry() {
+                if e.kind.limit_retries() {
+                    // We use a global counter since we don't have any local memory.
+                    let count = LIMIT_ERR_COUNT.fetch_add(1, Ordering::Relaxed);
+                    if count > LIMIT_ERR_MAX {
+                        // TODO: log details & cause?
+                        panic!("Too many RNG errors; last error: {}", e.msg());
+                    }
+                }
+                
+                if e.kind.should_wait() {
+                    #[cfg(feature="std")]{
+                        let dur = ::std::time::Duration::from_millis(10);
+                        ::std::thread::sleep(dur);
```

Comment:
Ideally, we could wait/poll on `/dev/random`. The systems that can return `WOULDBLOCK` also support polling on that device to wait for it to become ready. But that is way more complex...

Reply:
What if the RNG never becomes available? Would we effectively hang the application?

Reply:
Yes. But how long should we wait? I have no idea.

Reply:
Thinking about it some more, `/dev/(u)random` is not applicable. It would block during initialization, not during `fill_bytes`. I can't really imagine something that would return `ErrorKind::NotReady `...

Waiting for 1 second and 100 tries seems like an ok guess to me for a default function like this.
We should not wait to short: a delay of a fraction of a second is better than panicking.
We should not wait to long: it may cause something critical to hang.

It really depends on the cause for waiting, and what the applications wants. But then there always is `try_fill`, what someone using a RNG that can fail should really be using...

Reply:
I'm not very comfortable with a timeout, especially 1 second. Whatever caused the `NotReady` could easily take longer, e.g. OS startup.

Possibly a total delay of a minute with 100ms sleep intervals would be more appropriate?

Reply:
I don't think OS startup is something this has to handle, because that error is returned in `OsRng::new()`, not in `try_fill`

Reply:
For the current linux impl, yes. For other platforms though, is this something that is required?

Reply:
Am I reading this correctly, adding a delay of anything on the order of minute(s) seems crazy. I'm not sure I've ever used code that waited that long for my automatically.

Maybe I'm confused and off base here.

Reply:
I'm open to changing this — but it's handling a probably rare error, so not too concerned.

This is merged now so best open a new PR or issue to discuss.